app/models/validations/household_validations.rb

```diff
@@ -73,6 +73,15 @@ def validate_previous_housing_situation(record)
     end
   end
 
+  def validate_referral(record)
+    if record.rsnvac == "Relet to tenant who occupied same property as temporary accommodation" && REFERRAL_INVALID_TMP.include?(record.referral)
+      record.errors.add :referral, I18n.t("validations.household.referral.rsnvac_non_temp")
+    end
+    if record.tenancy == "Secure (including flexible)" && record.referral.present? && record.referral != "Internal transfer"
+      record.errors.add :referral, I18n.t("validations.household.referral.secure_tenancy")
+    end
```

Comment:
I wonder if you can set this up to use something like the case log constants instead of the text - it might make the code less brittle if you update the content of the answers

Reply:
I think we'll need to do that on all validations and tests at some point very soon (either use constants or the DB value directly). Wouldn't worry about that in this PR specifically though.

Reply:
Seems like the general pattern in our codebase right now. Might be worth fixing, is now the right time to do it? Maybe, but I'd like to discuss if it's a good use of time!

Reply:
@kosiakkatrina Based on Dan's reply we can save this task for a future PR (or PRs) 🙂